appjson/healthcheck.go

```diff
@@ -129,6 +138,31 @@ func (h Healthcheck) Execute(container types.ContainerJSON, ctx HealthcheckConte
 	return h.executeUptimeCheck(container)
 }
 
+func (h Healthcheck) HandleFailure(errors []error) []error {
+	if h.OnFailure == nil {
+		return nil
+	}
+	if len(h.OnFailure.Command) > 0 {
+		cmd := exec.Command(h.OnFailure.Command[0], h.OnFailure.Command[1:]...)
+		if err := cmd.Run(); err != nil {
+			return []error{fmt.Errorf("failed to execute on failure command: %s", err)}
+		}
+	}
+	if len(h.OnFailure.Url) > 0 {
+		data := map[string]interface{}{
+			"healthcheck_name": h.GetName(),
+			"errors":           errors,
+		}
+		json_data, err := json.Marshal(data)
+		if err != nil {
+			return []error{fmt.Errorf("failed to encode data as JSON: %s", err)}
+		}
+		if _, err = http.Post(h.OnFailure.Url, "application/json", bytes.NewBuffer(json_data)); err != nil {
```

Comment:
Does `net/http.Post()` return an error on non-200?

Reply:
I added the correct check